From 8482625b4a5bda52906b77d2991d95257fe403ad Mon Sep 17 00:00:00 2001
From: Steve Muckle <steve.muckle@linaro.org>
Date: Wed, 14 Oct 2015 17:39:55 -0700
Subject: [PATCH 16/19] sched/cpufreq_sched: return from kthread instead of
 do_exit()

Kthreads may simply return rather than calling do_exit().

Signed-off-by: Steve Muckle <steve.muckle@linaro.org>
---
 kernel/sched/cpufreq_sched.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/cpufreq_sched.c b/kernel/sched/cpufreq_sched.c
index db4b6fc..bd8a35c 100644
--- a/kernel/sched/cpufreq_sched.c
+++ b/kernel/sched/cpufreq_sched.c
@@ -106,7 +106,7 @@ static int cpufreq_sched_thread(void *data)
 		cpufreq_sched_try_driver_target(policy, gd->freq);
 	} while (!kthread_should_stop());
 
-	do_exit(0);
+	return 0;
 }
 
 static void cpufreq_sched_irq_work(struct irq_work *irq_work)
-- 
1.9.1

